action.py

```diff
@@ -24,14 +24,24 @@ def __str__(self):
     def __eq__(self, other): 
         return self.__dict__ == other.__dict__
 
-    def groundify(self, objects):
+    def groundify(self, objects, types):
         if not self.parameters:
             yield self
             return
         type_map = []
         variables = []
-        for var, type in self.parameters:
-            type_map.append(objects[type])
+        for var, typ in self.parameters:
+            type_stack = [typ]
```

Comment:
Here is the most important part.
If ``type == object`` then typemap appends all objects.
No need to consider 'object' in PDDL_Parser ``parse_types``.